core/types/transaction.go

```diff
@@ -36,11 +35,11 @@ import (
 var (
 	ErrInvalidSig               = errors.New("invalid transaction v, r, s values")
 	errNoSigner                 = errors.New("missing signing methods")
-	skipNonceDestinationAddress = map[string]bool{
-		common.XDCXAddr:                         true,
-		common.TradingStateAddr:                 true,
-		common.XDCXLendingAddress:               true,
-		common.XDCXLendingFinalizedTradeAddress: true,
+	skipNonceDestinationAddress = map[common.Address]struct{}{
```

Comment:
we should use `skipNonceDestinationAddress = map[common.Address]bool` to simplify function `IsSkipNonceTransaction` later.